### ISSUE TITLE
Incompatibility with "More_Emotes" and "CrossHair "

### ISSUE THREAD

**Issue by AJAL8000** [julian-perge/LC_OpenMonitors#4]
Description:
For some very strange reason. These two mods do not work when Open Monitors is active. I have no idea why this happens, because at first glance they have nothing in common.

Edit: Corporate Restructure doesn't have this problem btw. No idea why.

Mods:
->https://thunderstore.io/c/lethal-company/p/CTNOriginals/CrossHair/
->https://thunderstore.io/c/lethal-company/p/Sligili/More_Emotes/

I get these errors in the Log:

`[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
MoreEmotes.Scripts.CustomAnimationObjects.FindLegs () (at <e152e1967b684dcc9ff55d637143b2c4>:0)
MoreEmotes.Scripts.CustomAnimationObjects.Update () (at <e152e1967b684dcc9ff55d637143b2c4>:0)

[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
CrossHair.Patches.PlayerControllerBPatch.Update (GameNetcodeStuff.PlayerControllerB& __instance) (at <c365bbdacd8e49ef99d1784f90d21d27>:0)
(wrapper dynamic-method) GameNetcodeStuff.PlayerControllerB.DMD<GameNetcodeStuff.PlayerControllerB::Update>(GameNetcodeStuff.PlayerControllerB)`

It would be great if you could find out what causes this error. To clarify, Open_Monitors works fine, the problem is that for some reason it makes the other 2 mods stop working.

**Comment by julian-perge** [comment]
Thanks for the issue on this! I looked briefly into the Crosshair code and found the method mentioned here: https://github.com/CTN-Originals/LethalCompany-CrossHair/blob/main/CrossHair/Patches/PlayerControllerBPatch.cs#L127

Even looking at the code it's not immediately clear to me as to what line is causing the issue. I know when I was testing OpenMonitors, I had to add the delay after the CreditsMonitor triggered because if it tried to update immediately in the patch, it was causing null references. I imagine something like that is happening here as well with `CrossHair`.

**Comment by julian-perge** [comment]
For the `MoreEmotes` one, same findings as `CrossHair`. Reading the code it just seems like it's an issue with that mod, and not `OpenMonitors`. 

The `FindLegs()`  method is trying to find the legs on the players GameObject:

```csharp
private void FindLegs()
{
_legs = (_player.transform.Find("ScavengerModel").Find("LEGS").GetComponent<SkinnedMeshRenderer>();
}
```

**Comment by a1russell** [comment]
I am also experiencing this issue. I'm not seeing how it could be an issue with More Emotes and/or CrossHair, given that they work without OpenMonitors installed, and they also work when Corporate Restructure is installed.

If it helps, the issue is still present even with all of the terminals hidden via config options.

I don't know anything about C# code or modding, but might the issue have to do with how OpenMonitors is patching "PlayerControllerB"?

**Comment by CTNOriginals** [comment]
> https://github.com/CTN-Originals/LethalCompany-CrossHair/blob/main/CrossHair/Patches/PlayerControllerBPatch.cs#L127
> Even looking at the code it's not immediately clear to me as to what line is causing the issue. I know when I was testing OpenMonitors, I had to add the delay after the CreditsMonitor triggered because if it tried to update immediately in the patch, it was causing null references. I imagine something like that is happening here as well with `CrossHair`.

@julian-perge Could you please elaborate more on this and also point out where your have face this problem and how you have fixed it?

**Comment by julian-perge** [comment]
> > https://github.com/CTN-Originals/LethalCompany-CrossHair/blob/main/CrossHair/Patches/PlayerControllerBPatch.cs#L127
> > Even looking at the code it's not immediately clear to me as to what line is causing the issue. I know when I was testing OpenMonitors, I had to add the delay after the CreditsMonitor triggered because if it tried to update immediately in the patch, it was causing null references. I imagine something like that is happening here as well with `CrossHair`.
> 
> @julian-perge Could you please elaborate more on this and also point out where your have face this problem and how you have fixed it?

There's a `CoroutineHelper` class that you then pass an `IEnumerator` method to where you can use `yield` statements:

```csharp
    [HarmonyPostfix]
    [HarmonyPatch(nameof(GameNetcodeStuff.PlayerControllerB.ConnectClientToPlayerObject))]
    private static void OnPlayerConnect()
    {
        CoroutineHelper.Instance.StartCoroutine(WaitOnPlayerConnectForMonitorsToBeCreated());
    }

    private static IEnumerator WaitOnPlayerConnectForMonitorsToBeCreated()
    {
        ModLogger.LogDebug("WaitOnPlayerConnectForMonitorsToBeCreated");
        yield return new WaitUntil(() => CreditsMonitor.Instance && LifeSupportMonitor.Instance);
        CreditsMonitor.Instance.UpdateMonitor();
        LifeSupportMonitor.Instance.UpdateMonitor();
        PlayersLifeSupportMonitor.Instance.UpdateMonitor();
    }
```

**Comment by julian-perge** [comment]
Huh, I think I found the problem after digging a little deeper.

https://github.com/julian-perge/LC_OpenMonitors/blob/main/OpenMonitors/Monitors/Setup.cs#L100

Here I'm renaming the GameObject to `Deadline` from `HeaderText (1)`, but then in CrossHair, it's trying to look for the original object name which is `Header (1)`: https://github.com/CTN-Originals/LethalCompany-CrossHair/blob/main/Crosshair/Patches/HUDManagerPatch.cs#L22

**Comment by CTNOriginals** [comment]
> Huh, I think I found the problem after digging a little deeper.
> 
> https://github.com/julian-perge/LC_OpenMonitors/blob/main/OpenMonitors/Monitors/Setup.cs#L100
> 
> Here I'm renaming the GameObject to `Deadline` from `HeaderText (1)`, but then in CrossHair, it's trying to look for the original object name which is `Header (1)`: https://github.com/CTN-Originals/LethalCompany-CrossHair/blob/main/Crosshair/Patches/HUDManagerPatch.cs#L22

Yes, i have ndeed just found that same issue, the text i use as a reference is one of the monitors that you likely move around in some way which results in the crosshair mod not having the text reference. that is awkward...

**Comment by CTNOriginals** [comment]
Thanks for working with me on this bug and hunting down the issue, I will update my mod once I have time.

**Comment by julian-perge** [comment]
Should be fixed in v1.1.0 update. Sorry for the delay!
